File: EDAII x Python/Ordenamiento/ShakerSort.py
class ShakerSort:

	arr = []

	def __init__(self, arr):
		'Constructor de la clase'
		self.arr = arr

	def shakerSort(self):
		'Algoritmo de ordenamiento'
		n = len(self.arr)
		inf = 0
		sup = n-1

		while inf <= sup:
			# Pasada de Derecha a Izquierda
			for j in range(sup,inf,-1):
				if self.arr[j] < self.arr[j-1]:
					self.arr[j],self.arr[j-1] = self.arr[j-1],self.arr[j]	# Intercambio
			inf +=1

			for j in range(inf,sup):
				if self.arr[j] > self.arr[j+1]:
					self.arr[j],self.arr[j+1] = self.arr[j+1],self.arr[j]	# Intercambio
			sup -=1

# Ejecutando el objeto
programa = ShakerSort([8,3,7,4,9,5,2,1])
programa.shakerSort()
print(programa.arr)

File: EDAII x Java/Ordenamiento/ShakerSort.java
public class ShakerSort{

	int[] arr = {8,3,7,4,9,5,2,1};

	public static void main(String[] args){
		ShakerSort ss = new ShakerSort();
		ss.shakerSort(ss.arr); // Ordenamiento ShakerSort

		for (int i=0; i<ss.arr.length; i++){
			System.out.println(ss.arr[i]);
		}
	}

	public void shakerSort(int[] A){
		int n = A.length;
		int inf = 0, sup = n-1;

		while(inf <= sup){
			// Pasada de Derecha a Izquierda
			for(int j=sup; j>inf; j--){
				if (A[j] < A[j-1]){
					int tmp = A[j];
					A[j] = A[j-1];
					A[j-1] = tmp;
				}
			}
			inf++;

			// Pasada de Izquierda a Derecha
			for(int j=inf; j<sup; j++){
				if (A[j] > A[j+1]){
					int tmp = A[j];
					A[j] = A[j+1];
					A[j+1] = tmp;
				}
			}
			sup--;
		}
	}

}